src/System-Sources/CompiledMethod.extension.st

```diff
@@ -25,5 +25,6 @@ CompiledMethod >> putSource: source withPreamble: preambleBlock [
 			"Update with new source pointer"
 			self setSourcePointer: newSourcePointer ]
 		onFail: [
-			self becomeForward: (self copyWithSource: source) ]
+			self traceCr: 'Warning, could not write sources'.
```

Comment:
I'm not sure #traceCr: is part of the kernel so if this gets called during the beginning of the bootstrap it might crash. Maybe there should be a #respondsTo: if that can happen?

Reply:
But how can we log? logging is quite essential... else we never know what happens.

Reply:
I removed it

Reply:
We don't have the transcript but we have the stdio. 

Maybe we could find a way to use Transcript if present to log, and if it's not present log on Stdout?

Reply:
Hum... We do have the NonInteractiveTranscript. Maybe we could move the #traceCr: & co methods to this package so that we can log in the beginning of the bootstrap?